src/server/components/dashboard/listsItems/item/update/actionHandlers/archive.ts

```diff
@@ -0,0 +1,20 @@
+import { Request, Response } from "express";
+import { archiveListItem } from "server/models/listItem";
+
+export async function archive(req: Request, res: Response): Promise<void> {
+  const userId = req?.user?.userData?.id;
+  const { listItemUrl, listIndexUrl, listItem } = res.locals;
+  const { reason } = req.body;
+
+  try {
+    await archiveListItem(listItem.id, userId!, reason);
+
+    req.flash("successBannerTitle", `${listItem.jsonData.organisationName} has been archived`);
+    req.flash("successBannerHeading", "Archived");
+    req.flash("successBannerColour", "green");
+    return res.redirect(listIndexUrl);
+  } catch (error: any) {
+    req.flash("errorMsg", `${listItem.jsonData.organisationName} could not be archived. ${error.message}`);
```

Comment:
I know we do this in a few places but it's worth fixing from now on. It's likely this error comes from prisma. req.flash is for a user to see. It's unlikely they will know how to resolve a prisma error. 

It's better to just log the error message, and just show them "foo could not be archived".